src/Language/Haskell/GHC/ExactPrint/Preprocess.hs

```diff
@@ -185,7 +189,8 @@ getPreprocessedSrcDirectPrim cppOptions src_fn = do
   hsc_env <- GHC.getSession
   let dfs = GHC.extractDynFlags hsc_env
       new_env = GHC.replaceDynFlags hsc_env (injectCppOptions cppOptions dfs)
-  (dflags', hspp_fn) <- GHC.liftIO $ GHC.preprocess new_env (src_fn, Nothing)
+  (dflags', hspp_fn) <-
+      GHC.liftIO $ GHC.preprocess new_env (src_fn, Just (GHC.Cpp GHC.HsSrcFile))
```

Comment:
Unfortunately I don't know exactly. Without a test covering this case it's difficult to tell. I deduced that it was necessary from GHC's "documentation".